packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -0,0 +1,274 @@
+      setFilteredNodes(filterOptions(nodes, inputValue));
+    }
+  }, [value]);
+
+  const activeOptionRef = useRef();
+  const htmlId = useId('combobox');
+  const ignoreBlur = useRef(false);
+  const inputRef = useRef();
+  const containerRef = useRef();
+  const listboxRef = useRef();
+
+  useEffect(() => {
+    if (open && activeOptionRef.current) {
+      maintainScrollVisibility(activeOptionRef.current);
+    }
+  }, [activeIndex]);
+
+  const activeId = open ? `${htmlId}-${activeIndex}` : '';
+
+  const onInput = () => {
+    const curValue = inputRef.current.value;
+    setFilteredNodes(filterOptions(nodes, curValue));
+    setActiveIndex(0);
+    setSelectedIndex(null);
+
+    if (inputValue !== curValue) {
+      setInputValue(curValue);
+    }
+
+    if (!open) {
+      updateMenuState(true, false);
+    }
+  };
+
+  const onInputKeyDown = (event) => {
+    const { key } = event;
+    const max = filteredNodes.length - 1;
+    const action = getActionFromKey(key, open);
+
+    switch (action) {
+      case MenuActions.Next:
+      case MenuActions.Last:
+      case MenuActions.First:
+      case MenuActions.Previous:
+        event.preventDefault();
+        return onOptionChange(getUpdatedIndex(activeIndex, max, action));
+      case MenuActions.CloseSelect:
+        event.preventDefault();
+        onOptionSelect(activeIndex);
+        return;
+      case MenuActions.Close:
+        event.preventDefault();
+        return updateMenuState(false);
+      case MenuActions.Open:
+        return updateMenuState(true);
+      default:
+        return;
+    }
+  };
+
+  const onInputBlur = () => {
+    if (ignoreBlur.current) {
+      ignoreBlur.current = false;
+      return;
+    }
+
+    updateMenuState(false, false);
+  };
+
+  const onOptionChange = (index) => {
+    setActiveIndex(index);
+  };
+
+  const onOptionClick = (index) => {
+    onOptionChange(index);
+    onOptionSelect(index);
+  };
+
+  const onOptionMouseDown = () => {
+    ignoreBlur.current = true;
+  };
+
+  const onOptionSelect = (index) => {
+    const selected = filteredNodes[index];
+    if (selected) {
+      onChange(selected.props.value);
+      return updateMenuState(false);
+    }
+
+    if (creatable) {
+      onCreateOption(inputValue);
+      updateMenuState(false);
+    }
+  };
+
+  const updateMenuState = (open, callFocus = true) => {
+    setOpen(open);
+    callFocus && inputRef.current.focus();
+  };
+
+  const filteredNodesClone = Children.toArray(filteredNodes).map((node, i) => {
+    const isActive = activeIndex === i;
+    return cloneElement(node, {
+      id: `${htmlId}-${i}`,
+      'aria-selected': selectedIndex === i ? 'true' : false,
+      ref: (r) => {
+        if (isActive) activeOptionRef.current = r;
+      },
+      onClick: () => onOptionClick(i),
+      onMouseDown: onOptionMouseDown,
+      isSelected: isActive,
+    });
+  });
+
+  return (
+    <>
+      <MainRow ref={containerRef} disabled={disabled}>
```

Comment:
The disabled props will spread on the underlying DOM and we don't really want that 😓